server/common/HatoholArmPluginInterface.cc

```diff
@@ -157,6 +157,10 @@ struct HatoholArmPluginInterface::Impl {
 		connected = false;
 		hapi->onSessionChanged(NULL);
 	}
+	
+	bool connectstat(void) {
+		return connected;
```

Comment:
The member: connected has to be accessed with the lock: connectionLock in order to avoid races.